src/spikeinterface/core/numpyextractors.py

```diff
@@ -457,7 +457,8 @@ def from_peaks(peaks, sampling_frequency, unit_ids) -> "NumpySorting":
         spikes["sample_index"] = peaks["sample_index"]
         spikes["unit_index"] = peaks["channel_index"]
         spikes["segment_index"] = peaks["segment_index"]
-
+        order = np.argsort(spikes["sample_index"])
+        spikes = spikes[order]
```

Comment:
This is wrong!!!!!!!
You have to sort by segment index and then sample index.

Reply:
Good catch, always forgetting the segment. This should now work with lexsort